kolibri/plugins/coach_tools/assets/src/actions.js

```diff
@@ -16,35 +17,18 @@ const Constants = require('./state/constants');
 const logging = require('kolibri.lib.logging');
 
 
-/* returns an array of the values of an object */
-function _vals(obj) {
-  return Object.entries(obj).map(([key, value]) => value);
-}
-
-
-/* Only certain types of parameter updates require the 'loading' flag to be set */
-function _useReportPageLoadingFlag(newParams, oldParams) {
-  if (!newParams || !oldParams) {
-    return true;
-  }
-  if (Object.entries(newParams).length !== Object.entries(newParams).length) {
-    return true;
+/* find the keys that differ between the old and new params */
+function _diffKeys(newParams, oldParams) {
+  if (!oldParams) {
+    return Object.entries(newParams).map(([key, value]) => key);
```

Comment:
enumerate actually returns values, whereas this one returns keys.

In both cases, I think I'll switch to lodash.